torcharrow/test/test_text_ops.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import json
+import unittest
+from functools import lru_cache
+from pathlib import Path
+
+
+import torcharrow as ta
+import torcharrow._torcharrow as _ta
+import torcharrow.dtypes as dt
+import torchtext.transforms as T
+from torcharrow import functional
+
+
+_ASSET_DIR = (Path(__file__).parent / "asset").resolve()
+
+
+def get_asset_path(*path_components):
+    """Get the path to the file under `test/assets` directory."""
+    return str(_ASSET_DIR.joinpath(*path_components))
+
+
+@lru_cache()
+def bytes_to_unicode():
+    """
+    Original Source: https://github.com/openai/gpt-2/blob/master/src/encoder.py#L9
+
+    Returns list of utf-8 byte and a corresponding list of unicode strings.
+    The reversible bpe codes work on unicode strings.
+    This means you need a large # of unicode characters in your vocab if you want to avoid UNKs.
+    When you're at something like a 10B token dataset you end up needing around 5K for decent coverage.
+    This is a signficant percentage of your normal, say, 32K bpe vocab.
+    To avoid that, we want lookup tables between utf-8 bytes and unicode strings.
+    And avoids mapping to whitespace/control characters the bpe code barfs on.
+    """
+    bs = (
+        list(range(ord("!"), ord("~") + 1))
+        + list(range(ord("¡"), ord("¬") + 1))
+        + list(range(ord("®"), ord("ÿ") + 1))
+    )
+    cs = bs[:]
+    n = 0
+    for b in range(2**8):
+        if b not in bs:
+            bs.append(b)
+            cs.append(2**8 + n)
+            n += 1
+    cs = [chr(n) for n in cs]
+    return dict(zip(bs, cs))
+
+
+# copied from GPT2BPETokenizer __init__ method
+def init_bpe_encoder():
+    encoder_json_path = get_asset_path("gpt2_bpe_encoder.json")
+    vocab_bpe_path = get_asset_path("gpt2_bpe_vocab.bpe")
```

Comment:
I wonder if instead of creating these local assets in the TA repo, we could directly rely on downloading it from[ pytorch models repo](https://s3.console.aws.amazon.com/s3/buckets/pytorch?region=us-east-1&prefix=models/text/&showversions=false)? In other words, we could use a temporary directory for testing and use [tochhub API](https://pytorch.org/docs/stable/hub.html#torch.hub.download_url_to_file) to download asset locally for testing.

Reply:
I can give this a shot. The only thing I worry about is that adding a test that requires network access can be problematic when enabling the test within fbcode and we may have to rely on monkey patching to make it work as we do in torchtext. In addition, tests requiring downloads tend to increate the time it takes for all tests to pass in CI. This was one of the reasons we introduced mocking in torchtext to get away from test downloads. Wdyt? 

Also @wenleix lmk if you have an opinion here

Reply:
> The only thing I worry about is that adding a test that requires network access can be problematic when enabling the test within fbcode

Ahh, that's a good point. I didn't think about it when making the proposal. Ya, in that case I think let's go with the asset files. I remember there was discussion on adding a small size vocab for GPT2 (similar to mocking) instead of full model based vocab. Unless @wenleix  see an issue with the file size, I think we are good to go with the current solution :)

Reply:
Sounds good!